### ISSUE TITLE
Too slow source error message on Nvidia Shield TV

### ISSUE THREAD

**Issue by gsantosoliver** [cguZZman/plugin.googledrive#67]
**Describe the bug**
This is a continuation of #54 since new facts about the issue have not been addressed due to the fact the original issue is closed.

I have 2 devices running this add-on:

1. OnePlus 6 with latest Android 9 and latest Kodi and latest add-on version.
2. Shield TV 2017 with all the latest Android/Firmware running the latest Kodi and latest add-on version. This is my main media player - it is the main movie player in my Home Theater. 

I use the same userid to connect the GDrive add-on o both devices. 

I have both connected to the same Asus Rog Rapture Router - the smartphone using AC Wireless Network and Shield TV with Gigabit cable. The router is connected to the ISP ONT. 

Currently I've got 200mbps down 100 mbps up connection to the internet.

I run any video on the Smartphone without any problem, any delay/stop. Things just works great there! 
  On the Shield TV I got the "Slow connection to play/stream" error every other 3 minutes or so! I can't even use it anymore.

What can I do to make the Shield TV problem go away?! 

**To Reproduce**
1. Setup the GDrive Add-on from the Kodi Official Repository on Kodi Leia (18.1) in the Nvidia Shield TV running the latest firmware version.
2. Play any 4K video from the GDrive movie source you have just set up. 
3. After a couple of minutes the video freezes and you get a cache icon with a loading percentage icon that never goes goes beyond 0%

**Your debug log file**
1. Here is the log using Aura skin and wireless (I thought it may have been the cable, don't know - just want to use the same wireless AC as the smartphone to test) connection to the internet: https://paste.kodi.tv/nekuyihono.kodi
 
2. Here is the log after changing the skin to the default one and connecting Shield again to the router using the cable: https://paste.kodi.tv/wasuvomubo.kodi
 
3. And here is the log from my phone - I played the same movie for about 25 minutes without having ANY issues: https://paste.kodi.tv/okoxepajap.kodi
 
4. Here is the media information of the movie I used to reproduce the error: https://paste.kodi.tv/yovapogigo

**Screenshots**
To add later in case it is required - maybe

**Here is what we know so far**
Check this out:

1. I have uninstalled Kodi 18.1 completely from Shield TV and reinstalled from scratch. The only add-on I had running was the GDrive and its dependencies, the problem happened.

2. I removed Kodi 18.1 again and installed the version 18.0 64 bits for Android (Shield TV is 64 bits). Installed the plugin, set it up and ran the same movie. **It works flawlessly.**

Here is the log one running it on v18.0 without any problem in Shield: https://paste.kodi.tv/anecovuluj.kodi

**To sum up**
1. The problem does not happen on Kodi 18.1 32 bit for Android TV (my phone's case).
2. The problem happens on Kodi 18.1 64 bit for Android TV (NVidia Shield TV's case).
3. The problem does not happen on Kodi 18.0 64 bit for Android TV (NVidia Shield TV's case).
Would you please check?

**Comment by gsantosoliver** [comment]
Any updates on this one? :)

**Comment by GGATEV** [comment]
I have exactly the same behavior in higher versions of Kodi 18.0. If necessary, upload log files from Shield Tv

**Comment by gsantosoliver** [comment]
> **Describe the bug**
> This is a continuation of #54 since new facts about the issue have not been addressed due to the fact the original issue is closed.
> 
> I have 2 devices running this add-on:
> 
> 1. OnePlus 6 with latest Android 9 and latest Kodi and latest add-on version.
> 2. Shield TV 2017 with all the latest Android/Firmware running the latest Kodi and latest add-on version. This is my main media player - it is the main movie player in my Home Theater.
> 
> I use the same userid to connect the GDrive add-on o both devices.
> 
> I have both connected to the same Asus Rog Rapture Router - the smartphone using AC Wireless Network and Shield TV with Gigabit cable. The router is connected to the ISP ONT.
> 
> Currently I've got 200mbps down 100 mbps up connection to the internet.
> 
> I run any video on the Smartphone without any problem, any delay/stop. Things just works great there!
> On the Shield TV I got the "Slow connection to play/stream" error every other 3 minutes or so! I can't even use it anymore.
> 
> What can I do to make the Shield TV problem go away?!
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Setup the GDrive Add-on from the Kodi Official Repository on Kodi Leia (18.1) in the Nvidia Shield TV running the latest firmware version.
> 2. Play any 4K video from the GDrive movie source you have just set up.
> 3. After a couple of minutes the video freezes and you get a cache icon with a loading percentage icon that never goes goes beyond 0%
> 
> **Your debug log file**
> 
> 1. Here is the log using Aura skin and wireless (I thought it may have been the cable, don't know - just want to use the same wireless AC as the smartphone to test) connection to the internet: https://paste.kodi.tv/nekuyihono.kodi
> 2. Here is the log after changing the skin to the default one and connecting Shield again to the router using the cable: https://paste.kodi.tv/wasuvomubo.kodi
> 3. And here is the log from my phone - I played the same movie for about 25 minutes without having ANY issues: https://paste.kodi.tv/okoxepajap.kodi
> 4. Here is the media information of the movie I used to reproduce the error: https://paste.kodi.tv/yovapogigo
> 
> **Screenshots**
> To add later in case it is required - maybe
> 
> **Here is what we know so far**
> Check this out:
> 
> 1. I have uninstalled Kodi 18.1 completely from Shield TV and reinstalled from scratch. The only add-on I had running was the GDrive and its dependencies, the problem happened.
> 2. I removed Kodi 18.1 again and installed the version 18.0 64 bits for Android (Shield TV is 64 bits). Installed the plugin, set it up and ran the same movie. **It works flawlessly.**
> 
> Here is the log one running it on v18.0 without any problem in Shield: https://paste.kodi.tv/anecovuluj.kodi
> 
> **To sum up**
> 
> 1. The problem does not happen on Kodi 18.1 32 bit for Android TV (my phone's case).
> 2. The problem happens on Kodi 18.1 64 bit for Android TV (NVidia Shield TV's case).
> 3. The problem does not happen on Kodi 18.0 64 bit for Android TV (NVidia Shield TV's case).
>    Would you please check?

I am having the exact same problem with Mi Box S as well

**Comment by cguZZman** [comment]
I think you have an issue ready to be send to Kodi team. As I told you before, the addon does not play the file, Kodi does.
If you want to make an better case with them, you could store you movie file in a secured webserver (https) and then play it with Kodi from there. In this test scenario there won't be addon involved and the point is to prove that you are getting the same result.

**Comment by gsantosoliver** [comment]
@cguZZman - this is exactly why I've opened this issue here: I can play the same file using a External HD OR using a NFS or SMB server in Kodi 18.1! That makes me rule out Kodi 18.1 as the root cause and point to whatever else is used to play that file :)

Please reopen this one!

**Comment by cguZZman** [comment]
Nfs or smb is not https. Do the test on a secured webserver and we can see.

**Comment by gsantosoliver** [comment]
> Nfs or smb is not https. Do the test on a secured webserver and we can see.

I'll set up a secured https server on my local NAS and run it today. Please keep this opened. ;)

**Comment by cguZZman** [comment]
It has to be remote, with some latency, like with google drive servers. 
I think the easy way is to create a shared link in google drive and then play it directly in Kodi.

**Comment by gsantosoliver** [comment]
> It has to be remote, with some latency, like with google drive servers.
> I think the easy way is to create a shared link in google drive and then play it directly in Kodi.

Ok, I can try that. Where I am is 4:27 PM - I will do that test this evening and type the results here. Thanks, @cguZZman

**Comment by gsantosoliver** [comment]
@cguZZman - I got the URL to the movie - do you know how to play it on kodi? Thanks.

**Comment by cguZZman** [comment]
One way is to create a local strm file with the url inside. Then play the strm file in kodi.
Make sure you url is downloadable. Not just a link which redirects to a page.

**Comment by gsantosoliver** [comment]
> One way is to create a local strm file with the url inside. Then play the strm file in kodi.
> Make sure you url is downloadable. Not just a link which redirects to a page.

Hi @cguZZman - not sure if you have tried that - here it did not work - the url that is shared is https://drive.google.com/file/d/1yL0OyruKByxPtxTZ9yWrHKOHw1X6IOLd

Log https://paste.kodi.tv/erusikojim.kodi 

Am I missing anything?
